tests/test_base.cpp

```diff
@@ -562,7 +580,7 @@ std::string server_node::config_path() const
 	return m_path;
 }
 
-server_config server_node::config() const
+server_config &server_node::config()
```

Comment:
What does that mean? So far you haven't even described why is this patch needed and terminology used here is quite vague.
Please describe it in more details: why is it needed, what is this, why do you want backend 'serialization', what are you trying to achieve with these changes?

So far it looks more like a quick dirty hack to add some ugly pieces needed for obscure or no purpose.